lib/minstall.js

```diff
@@ -34,48 +39,28 @@ function logIfInRoot(message) {
   }
 }
 
-function logVersionConflict(messages, moduleFolderName) {
-
-  const moduleFolder = path.join(cwd, moduletools.modulesFolder, moduleFolderName);
-  console.log(`
-
-| UNCRITICAL DEPENDENCY-CONFLICTS FOUND:
-|   ${messages.join('\n|   ')}
-|
-| This is not an error, but consider using compatible package-versions
-| throughout the whole project.`);
-}
-
 function getLocalPackageInfo() {
-  return ModuleInfo.loadFromFolder('', '');
-}
-
-function getInstalledModules() {
-  return moduletools.getModules('node_modules');
-}
-
-function removeModulesFromPackageList(packageNames, moduleInfos) {
-  const moduleFolders = moduleInfos.map((moduleInfo) => {
-    return moduleInfo.folderName;
-  });
-
-  return packageNames.filter((packageName) => {
-    if (moduleFolders.indexOf(packageName) >= 0) {
-      logger.debug(`skipping ${packageName}`);
-    }
-    return !(moduleFolders.indexOf(packageName) >= 0);
-  });
+  return ModuleInfo.loadFromFolder(cwd, '');
 }
 
 function checkStartConditions() {
   logger.debug('checking start conditions');
   return Promise.all([
     systools.verifyFolderName(cwd, 'node_modules'),
     getLocalPackageInfo(),
+    systools.runCommand('npm --version'),
   ])
     .then((results) => {
       const folderName = results[0];
       localPackage = results[1];
+      npmVersion = results[2];
```

Comment:
If you would have used a map, you would not get this ugly integer-access to array fields.

Additionally it would be much more robust, as it does not depend on the order of the list.

Reply:
using a map here is difficult. `Promise.all()` wants an Array as input, and has an array with the results in the same order as the input as result